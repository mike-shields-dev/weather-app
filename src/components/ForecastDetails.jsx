import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/ForecastDetails.module.css";
import FormattedDate from "./FormattedDate";
import WindDirectionIcon from "./WindDirectionIcon";

const ForecastDetails = ({ date, temperature, humidity, wind }) => {
  return (
    <div className={styles["forecast-details"]} data-testid="forecast-details">
      <h2
        className={styles["forecast-details__date"]}
        data-testid="forecast-details-date"
      >
        <FormattedDate date={date} />
      </h2>
      <p
        className={styles["forecast-details__temperature-max"]}
        data-testid="forecast-details-temperature-max"
      >
        Max Temperature: <span>{temperature.max}</span> °C
      </p>
      <p
        className="forecast-details__temperature-min"
        data-testid="forecast-details-temperature-min"
      >
        Min Temperature: <span>{temperature.min}</span> °C
      </p>
      <p
        className="forecast-details__humidity"
        data-testid="forecast-details-humidity"
      >
        Humidity: <span>{humidity}</span>%
      </p>
      <p
        className={styles["forecast-details__wind"]}
        data-testid="forecast-details-wind"
      >
        <span>{wind.speed}</span>mph{" "}
        <WindDirectionIcon direction={wind.direction} />
      </p>
    </div>
  );
};

ForecastDetails.propTypes = {
  date: PropTypes.number.isRequired,
  temperature: PropTypes.shape({
    max: PropTypes.number,
    min: PropTypes.number,
  }).isRequired,
  wind: PropTypes.shape({
    speed: PropTypes.number,
    direction: PropTypes.string,
  }).isRequired,
  humidity: PropTypes.number.isRequired,
};

export default ForecastDetails;
