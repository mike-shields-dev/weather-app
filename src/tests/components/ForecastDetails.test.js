import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";
import unixTimestampToFormattedDate from "../../helpers/unixTimestampToFormattedDate";

const validProps = {
  date: 1525046400000,
  temperature: {
    max: 11,
    min: 4,
  },
  wind: {
    speed: 13,
    direction: "s",
  },
  humidity: 30,
  description: "Clear",
  icon: 800,
};

const { date, temperature, humidity, wind } = validProps;
const formattedDate = unixTimestampToFormattedDate(date);

describe("ForecastDetails", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ForecastDetails {...validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders formatted date, min and max temperature, humidity and wind speed and direction", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails {...validProps} />
    );

    expect(getByText(formattedDate)).toBeInTheDocument();
    expect(getByText(humidity)).toBeInTheDocument();
    expect(getByText(temperature.max)).toBeInTheDocument();
    expect(getByText(temperature.min)).toBeInTheDocument();
    expect(getByText(wind.speed)).toBeInTheDocument();
    expect(getByTestId("wind-direction-icon")).toBeInTheDocument();
  });

  it("renders elements with correct class names", () => {
    const { getByTestId } = render(<ForecastDetails {...validProps} />);

    expect(getByTestId("forecast-details-date")).toHaveClass(
      "forecast-details__date"
    );
    expect(getByTestId("forecast-details-humidity")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByTestId("forecast-details-temperature-max")).toHaveClass(
      "forecast-details__temperature-max"
    );
    expect(getByTestId("forecast-details-temperature-min")).toHaveClass(
      "forecast-details__temperature-min"
    );
    expect(getByTestId("forecast-details-wind")).toHaveClass(
      "forecast-details__wind"
    );
  });
});
