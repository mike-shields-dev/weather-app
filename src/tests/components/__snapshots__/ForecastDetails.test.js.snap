// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <h2
      class="forecast-details__date"
      data-testid="forecast-details-date"
    >
      <span>
        Mon 30th Apr
      </span>
    </h2>
    <p
      class="forecast-details__temperature-max"
      data-testid="forecast-details-temperature-max"
    >
      Max Temperature: 
      <span>
        11
      </span>
       °C
    </p>
    <p
      class="forecast-details__temperature-min"
      data-testid="forecast-details-temperature-min"
    >
      Min Temperature: 
      <span>
        4
      </span>
       °C
    </p>
    <p
      class="forecast-details__humidity"
      data-testid="forecast-details-humidity"
    >
      Humidity: 
      <span>
        30
      </span>
      %
    </p>
    <p
      class="forecast-details__wind"
      data-testid="forecast-details-wind"
    >
      <span>
        13
      </span>
      mph 
      <svg
        class="wind-direction-icon"
        data-testid="wind-direction-icon"
        fill="currentColor"
        style="transform: rotate(180deg);"
        version="1.1"
        viewBox="0 0 490 490"
        x="0px"
        y="0px"
      >
        <title>
          arrow pointing south
        </title>
        <path
          d="M480.086,490L245,339.229L9.914,490L245,0L480.086,490z"
        />
      </svg>
    </p>
  </div>
</DocumentFragment>
`;
